producer-c/producer-cloudwatch-integ/KvsProducerSampleCloudwatch.cpp

```diff
@@ -11,14 +11,14 @@ VOID sigintHandler(INT32 sigNum)
 // add frame pts, frame index, original frame size, CRC to beginning of buffer
 VOID addCanaryMetadataToFrameData(PFrame pFrame) {
     PBYTE pCurPtr = pFrame->frameData;
-    putUnalignedInt64(pCurPtr, pFrame->presentationTs / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
+    putUnalignedInt64((PINT64)pCurPtr, pFrame->presentationTs / HUNDREDS_OF_NANOS_IN_A_MILLISECOND);
     pCurPtr += SIZEOF(UINT64);
-    putUnalignedInt32(pCurPtr, pFrame->index);
+    putUnalignedInt32((PINT32)pCurPtr, pFrame->index);
     pCurPtr += SIZEOF(UINT32);
-    putUnalignedInt32(pCurPtr, pFrame->size);
+    putUnalignedInt64((PINT32)pCurPtr, pFrame->size);
     pCurPtr += SIZEOF(UINT32);
-    MEMSET(pCurPtr, 0x00, SIZEOF(UINT64));
-    putUnalignedInt64(pCurPtr, COMPUTE_CRC32(pFrame->frameData, pFrame->size));
+    MEMSET(pCurPtr, 0x00, SIZEOF(UINT32));
```

Comment:
Sorry, one more thing.. We don't really need the memset right? The next line will actually overwrite pCurPtr with the checksum anyway. I am not sure why this was there in the first place

Reply:
You are right. I had added it at the time for accuracy. 
Also, it looks like putUnalignedInt64 leads to a different value and putInt64 leads to a different value.

Reply:
For some reason, the end to end latency leads to very large values (negative) when I use putUnalignedInt64 but works fine when I use putInt64.

Reply:
There shouldn't be any difference. If we don't get a bus error (on platforms that don't support alignment) then we should be fine. There is a potential for endianness mismatch perhaps. Can we debug into this? This is a straight forward issue really